src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java

```diff
@@ -37,12 +37,24 @@ public BigQueryDataExtractionService(Storage storage) {
     this.storage = storage;
   }
 
-  void extract(BigQueryExtractionData extractionData) {
+  boolean extract(BigQueryExtractionData extractionData) {
+    if (extractionData.isExtracted()) {
```

Comment:
Why do we now need to keep track of whether this has been extracted and cleaned before?